promql/promqltest/testdata/functions.test

```diff
@@ -1022,6 +1022,13 @@ eval instant at 1m last_over_time(data[2m])
 	data{type="some_nan3"} 1
 	data{type="only_nan"} NaN
 
+eval instant at 1m avg_over_time(data[2m])
+	data{type="some_nan3"} 0.5
+	data{type="numbers"} 1.6666666666666667
+	data{type="some_nan"} 1
```

Comment:
DEMONSTRATES BUG related to original issue: [this test failure](https://github.com/prometheus/prometheus/actions/runs/12009023943/job/33472873752?pr=15451#step:6:246) proves that `avg_over_time` over [[2, 0, NaN]](https://github.com/prometheus/prometheus/pull/15451/files#diff-f277c3549a17549dbc047c4f13cdeebd450a0a69b2a568f7dde50935583b3540R999) produces `NaN` according to former (and still current) behavior instead of `1` expected on this line